control/preimage/src/treasury_commands.rs

```diff
@@ -0,0 +1,221 @@
+    },
+    Spend {
+        name: "Milestone completion reward #5",
+        asset: TreasuryAsset::DOT(12641),
+        delay: Some(120 * DAYS),
+    },
+    Spend {
+        name: "Milestone completion reward #6",
+        asset: TreasuryAsset::DOT(12641),
+        delay: Some(150 * DAYS),
+    },
+    Spend {
+        name: "Milestone completion reward #7",
+        asset: TreasuryAsset::DOT(12641),
+        delay: Some(180 * DAYS),
+    },
+    Spend {
+        name: "Milestone completion reward #8",
+        asset: TreasuryAsset::DOT(12641),
+        delay: Some(210 * DAYS),
+    },
+    Spend {
+        name: "Milestone completion reward #9",
+        asset: TreasuryAsset::DOT(12641),
+        delay: Some(240 * DAYS),
+    },
+    Spend {
+        name: "Milestone completion reward #10",
+        asset: TreasuryAsset::DOT(12641),
+        delay: Some(270 * DAYS),
+    },
+    Spend {
+        name: "Milestone completion reward #11",
+        asset: TreasuryAsset::DOT(12641),
+        delay: Some(300 * DAYS),
+    },
+    Spend {
+        name: "Milestone completion reward #12",
+        asset: TreasuryAsset::DOT(12641),
+        delay: Some(330 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #1",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(90 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #2",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(120 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #3",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(150 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #4",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(180 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #5",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(210 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #6",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(240 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #7",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(270 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #8",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(300 * DAYS),
+    },
+    Spend {
+        name: "General Incentive reward #9",
+        asset: TreasuryAsset::USDC(312500),
+        delay: Some(330 * DAYS),
+    },
+];
+
+pub const LAUNCH_BLOCK: u32 = 21292000;
+
+pub fn treasury_proposal(params: &TreasuryProposal2024Args) -> RelayRuntimeCall {
+    let mut calls: Vec<RelayRuntimeCall> = vec![];
+
+    for spend in SPENDS.iter() {
+        let (asset_id, asset_location, asset_amount) = spend.asset.into();
+        let call = make_treasury_spend(
+            params.beneficiary.into(),
+            asset_location,
+            asset_amount,
+            spend.delay.map(|delay| LAUNCH_BLOCK + delay),
```

Comment:
hm this will be a block in the past for the immediate spend. is that an issue?

Reply:
With if the spend delay is `None`, then the `map` is short-circuited (isnt executed), and so the final delay is `None`, which is immediate.